### PR TITLE
feat(aws-secrets-manager): migrate to secrets manager plugin event

### PR DESCRIPTION
**Pull request by overbit**
Fixes: amplication/amplication#7417

## PR Details

This PR update the aws secrets manager plugin to use the secretsmanager event from DSG v2.2.0

## PR Checklist

- [ ] Tests for the changes have been added
- [x] `npm build` doesn't throw any error
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: plugins/secrets-manager-aws-secret-manager/src/index.ts [hunk 1]
```diff
@@ -1,69 +1,86 @@
...
 
-    async beforeCreateServer(context: DsgContext, _: CreateServerParams, modules: ModuleMap): Promise<ModuleMap> {
-        const { fetchMode, secretNames } = getPluginSettings(context.pluginInstallations)
-        const staticPath = resolve(__dirname, "static", fetchMode.toLowerCase())
+  async beforeCreateServer(
+    context: DsgContext,
+    _: CreateServerParams,
+    modules: ModuleMap,
+  ): Promise<ModuleMap> {
+    const { fetchMode, secretNames } = getPluginSettings(
```
Comment (mulygottlieb) [inline plugins/secrets-manager-aws-secret-manager/src/index.ts:58]: Are you using secretNames anywhere in this method?

File: plugins/secrets-manager-aws-secret-manager/src/static/startup/providers/secrets/secretsManagerFactory.ts [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
+    var secrets: Partial<Record<Secrets, unknown>> = {};
 
     for (const path of Object.values(Secrets)) {
-      const [secret_id, secret_name] = path.split(":")
+      const [secret_id, secret_name] = path.split(":");
 
       try {
-        const response = await client.send(new GetSecretValueCommand({ SecretId: secret_id }))
+        const response = await client.send(
+          new GetSecretValueCommand({ SecretId: secret_id }),
```
Comment (mulygottlieb) [inline plugins/secrets-manager-aws-secret-manager/src/static/startup/providers/secrets/secretsManagerFactory.ts:25]: the comma at the end of line 25 seems odd

Comment (overbit) [reply plugins/secrets-manager-aws-secret-manager/src/static/startup/providers/secrets/secretsManagerFactory.ts:25]: gone

**Review by abrl91 — APPROVED** [review]

File: plugins/secrets-manager-aws-secret-manager/src/index.ts (hunk 1 above)
Comment (mulygottlieb) [reply plugins/secrets-manager-aws-secret-manager/src/index.ts]: I don't think it is gone @overbit

**Pull request merged**